client/src/app/tabs/dmn/modeler/features/decision-table-keyboard-bindings/__tests__/DecisionTableKeyboardBindingsSpec.js

```diff
@@ -0,0 +1,207 @@
+    const activeViewer = modeler.getActiveViewer();
+
+    const decisionTableKeyboardBindings = activeViewer.get('decisionTable').decisionTableKeyboardBindings;
+
+    const event = createKeyEvent('Z', { ctrlKey: true });
+
+    // when
+    decisionTableKeyboardBindings.handleKeydown(event);
+
+    // then
+    expect(undoSpy).to.have.been.called;
+  });
+
+
+  it('should redo', function() {
+
+    // given
+    const redoSpy = spy();
+
+    const modeler = createModeler({
+      commandStack: {
+        canRedo: () => true,
+        redo: redoSpy
+      },
+      eventBus: {
+        on: (_, callback) => callback()
+      },
+      decisionTable: {
+        _container: container
+      }
+    });
+
+    const activeViewer = modeler.getActiveViewer();
+
+    const decisionTableKeyboardBindings = activeViewer.get('decisionTable').decisionTableKeyboardBindings;
+
+    const event = createKeyEvent('Y', { ctrlKey: true });
+
+    // when
+    decisionTableKeyboardBindings.handleKeydown(event);
+
+    // then
+    expect(redoSpy).to.have.been.called;
+  });
+
+
+  it('should handle selectCellBelow', function() {
+
+    // given
+    const triggerSpy = spy();
+
+    const modeler = createModeler({
+      editorActions: {
+        trigger: triggerSpy,
+      },
+      eventBus: {
+        on: (_, callback) => callback()
+      },
+      decisionTable: {
+        _container: container
+      }
+    });
+
+    const activeViewer = modeler.getActiveViewer();
+
+    const decisionTableKeyboardBindings = activeViewer.get('decisionTable').decisionTableKeyboardBindings;
+
+    const selectableAncestorSpy =
+      sinon.stub(decisionTableKeyboardBindings, '_hasSelectableAncestor').returns(true);
+
+    const event = createKeyEvent('Enter', { ctrlKey: false });
+
+    // when
+    decisionTableKeyboardBindings.handleKeydown(event);
+
+    // then
+    expect(selectableAncestorSpy).to.have.been.called;
+    expect(triggerSpy).to.have.been.calledWith('selectCellBelow');
+
+    sinon.restore();
+
+  });
+
+
+  it('should handle selectCellAbove', function() {
+
+    // given
+    const triggerSpy = spy();
+
+    const modeler = createModeler({
+      editorActions: {
+        trigger: triggerSpy,
+      },
+      eventBus: {
+        on: (_, callback) => callback()
+      },
+      decisionTable: {
+        _container: container
+      }
+    });
+
+    const activeViewer = modeler.getActiveViewer();
+
+    const decisionTableKeyboardBindings = activeViewer.get('decisionTable').decisionTableKeyboardBindings;
+
+    const selectableAncestorSpy =
+      sinon.stub(decisionTableKeyboardBindings, '_hasSelectableAncestor').returns(true);
+
+    const event = createKeyEvent('Enter', {
+      ctrlKey: false,
+      shiftKey: true
+    });
+
+    // when
+    decisionTableKeyboardBindings.handleKeydown(event);
+
+    // then
+    expect(selectableAncestorSpy).to.have.been.called;
+    expect(triggerSpy).to.have.been.calledWith('selectCellAbove');
+
+    sinon.restore();

```

Comment:
This might never be called if an error is thrown before. Anyway, we probably don't need it as we don't spy on any other objects' methods.

Reply:
That could probably be right, but to make sure that we will not run into bugs in the future, I will move the restoring to an `afterEach` hook.